plugins/automod.py

```diff
@@ -273,7 +273,7 @@ async def process_messages(msgs: Iterable[Message]) -> None:
                     if any(role.id in exempt_roles for role in msg.author.roles):
                         continue
                 if (rule := active_rules.get(index)) is not None:
-                    reason = "Automatic action: message matches pattern {}".format(index)
+                    reason = format("Automatic action: message matches pattern {}\n||{!i}||", index, msg.content)
```

Comment:
Please include some safeguard against the `reason` becoming too long to fit in various places. E.g. use match information in `match` to select a substring around the match.